lib/msf/core/exploit/jsobfu.rb

```diff
@@ -18,14 +19,20 @@ def initialize(info={})
     # @param js [String] JavaScript code
     # @param opts [Hash] obfuscation options
     #    * :iterations [FixNum] Number of times to obfuscate
+    #    * :preserved_identifiers [Array] An array of identifiers to preserve during obfuscation
     # @return [::Rex::Exploitation::JSObfu]
     #
     def js_obfuscate(js, opts={})
       iterations = (opts[:iterations] || datastore['JsObfuscate']).to_i
+      identifiers = (opts[:preserved_identifiers] || datastore['JsIdentifiers'] || '').split(',')
       obfu = ::Rex::Exploitation::JSObfu.new(js)
```

Comment:
`opts[:preserved_identifiers]` here is an array, but you treat it like a String and call `split(',')` on it. This causes its array to end up double-wrapped:

```
=> {:iterations=>1, :preserved_identifiers=>[["alert"]]}
```

Using `opts[:preserved_identifiers].join` should fix the problem.